packages/wallet/src/wallet.ts

```diff
@@ -327,4 +327,57 @@ export class Wallet {
         await this.db.saveSilentPaymentAddress(address);
         return address;
     }
+
+    async scanSilentBlock(silentBlock: SilentBlock): Promise<void> {
+        const dustLimit = 546;
+        const matchedUTXOs: Coin[] = [];
+    
+        const coinType = this.network.network.bech32 === 'bc' ? 0 : 1;
+        const scanKey = this.masterKey.derivePath(`m/352'/${coinType}'/0'/1'/0`);
+        const spendKey = this.masterKey.derivePath(`m/352'/${coinType}'/0'/0'/0`);
+        
+        const scanPrivateKey = scanKey.privateKey;
+        const scanPublicKey = scanKey.publicKey;
+        const spendPublicKey = spendKey.publicKey;
+    
+        for (const transaction of silentBlock.transactions) {
+            const outputs = transaction.outputs.filter((output) => output.value >= dustLimit);
```

Comment:
I don't think we should do this check. We should store every coin

Reply:
i followed this [post,](https://github.com/setavenger/BIP0352-light-client-specification) but we can remove it if you'd prefer.